### ISSUE TITLE
Changing digital screen colour mode updates pixels only once and never again.

### ISSUE THREAD

**Issue by Syvil** [wiremod/wire#2039]
Steps to reproduce:
1. Set the colour mode address on a digital screen.
ex: `DS[1048569] = 3`
2. Write an image to the screen.
ex: 
![image](https://user-images.githubusercontent.com/35163854/77021037-e5a86d80-6942-11ea-9133-6374a9c72664.png)
3. Set the colour mode address to a different mode.
ex: `DS[1048569] = 2`
![image](https://user-images.githubusercontent.com/35163854/77021174-40da6000-6943-11ea-9d78-5700c417f0a5.png)
Note: Displayed image is messed up as expected
4. Set the colour mode address back to the original mode
ex: `DS[1048569] = 3`
![image](https://user-images.githubusercontent.com/35163854/77021326-a0387000-6943-11ea-810d-ba198f77231b.png)
Note: The displayed image *should* have returned back to the original colours.

Additionally:
* Attempting to rewrite the same colour to a pixel does not update the pixel to the correct colour mode. (Unless a different colour is written to it first)
* Any pixel that has a new colour written to it displays in the correct colour mode.
* Any pixel that has been updated to a new colour between steps 3 and 4 will update to the correct colour mode after step 4.

**Comment by thegrb93** [comment]
The clk address needs to be switched too.

**Comment by Syvil** [comment]
That works as a solution, but I feel like the original issue is still a bug.

**Comment by Syvil** [comment]
Although, that method does mean the clk address needs to stay changed until the screen has updated before it can be changed back to 1 to continue displaying edits to the image.

**Comment by thegrb93** [comment]
No, I think drawing is edge triggered via the clk input. The fact it works the first time might be a bug though.

**Comment by thegrb93** [comment]
Or the clk is just all kinds of screwed up in it.

**Comment by Syvil** [comment]
I'm sorry if this question is dumb, but how did using highspeed to change clk (1048575) cause the image to be update?
All I could find under the ENT:WriteCell function for address 1048575 was
https://github.com/wiremod/wire/blob/be2cea9d6664e835169f6e13a5c7afd94601b543/lua/entities/gmod_wire_digitalscreen/init.lua#L259-L261
Anything else triggered in that function is triggered by changing the color mode (1048569) too, but that didn't update the image more than once.
I haven't written any code for gmod though, so I could just be be misunderstanding something here.

**Comment by thegrb93** [comment]
It looks like setting the color more shouldn't trigger a redraw at all. It should only affect newly set pixels.
